lib/MetaCPAN/Model/Tarball.pm

```diff
@@ -0,0 +1,167 @@
+package MetaCPAN::Model::Tarball;
+
+use Archive::Any;
+use CPAN::DistnameInfo ();
+use DateTime           ();
+use DDP;
+use Log::Contextual qw( :log :dlog );
+use MetaCPAN::Types qw(ArrayRef File Dir);
```

Comment:
`use MetaCPAN::Types qw(ArrayRef Dir File HashRef Str);`

Then you can do `isa => Str` (no quotes) etc.